### PR TITLE
Week 1 assessment jh

### PR DESCRIPTION
**Pull request by justinhilario**
finished all coding challenges with the expected outputs; however still a little confused on the overall logic of finding bigger numbers or longer strings within arrays, although I was able to figure out question 6.

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -9,67 +9,171 @@
 // --------------------1) Create a function that determines if a temperature is below boiling point, at boiling point or above boiling point. Boiling point is 212 degrees Fahrenheit.
 // Use the test variables provided below. Expected outcome: "85 is below boiling point" "350 is above boiling point" "212 is at boiling point"
 
+// my pseudocode: we want to create a function which will take in a temperature and create a decision that our temperature is one of three possibilities
+
 var temp1 = 85
 var temp2 = 350
 var temp3 = 212
 
+var randomTemp = Math.floor(Math.random()* 350)
```
Comment (minachuong) [inline code-challenges.js:18]: VERY COOL! EDGE CASE TESTING!!  🥇

File: code-challenges.js [hunk 2]
```diff
@@ -9,67 +9,171 @@
...
+
+const decideTemperature = (tempStatus) => {
+    if (tempStatus <= 211){
+        return `${tempStatus} is below boiling point.`
+    } else if (tempStatus === 212){
+        return `${tempStatus} is at boiling point.`
+    } else if (tempStatus > 212){
+        return `${tempStatus} is above boiling point.`
+    }
+}
```
Comment (minachuong) [inline code-challenges.js:28]: what if a number is not passed in?

File: code-challenges.js [hunk 3]
```diff
@@ -9,67 +9,171 @@
...
 
 // --------------------2) Create a function that multiplies each number in the array by 5 using a for loop.
 // Use the test variable provided below. Expected outcome: [15, 35, 0, 30, -45] [10, 25, -40, 100, 160]
 
+//my pseudocode: we want a function that will take each value of an array, determine each value within the array meets our conditions, and outputs those values multiplied by 5 in an array
+
 var myForNumbers1 = [3, 7, 0, 6, -9]
 var myForNumbers2 = [2, 5, -8, 20, 32]
 
+const multipliesByFive = (anArray) => {
```
Comment (minachuong) [inline code-challenges.js:50]: awesome descriptive function naming!

File: code-challenges.js [hunk 4]
```diff
@@ -9,67 +9,171 @@
...
+    let storageForNewArray = []
+    for (let i=0 ; i < anArray.length ; i++){
+        storageForNewArray.push(anArray[i]*5)
+    }
+    return storageForNewArray
```
Comment (minachuong) [inline code-challenges.js:55]: again great descriptive variable naming.

File: code-challenges.js [hunk 5]
```diff
@@ -9,67 +9,171 @@
...
 // --------------------3) Create a function that multiplies each number in the array by 5 using map.
 // Use the test variable provided below. Expected outcome: [435, 30, 140, 80, 10] [40, -35, 0, 30, 60]
 
+// my pseudocode: we want a function that will take each value of an array and use the built in .map to multiply each value of the array by 5
+
 var myMapNumbers1 = [87, 6, 28, 16, 2]
 var myMapNumbers2 = [8, -7, 0, 6, 12]
 
+const multipliesArrayValuesByFive = (theArrayValue) => {
+   return theArrayValue.map( value =>{
```
Comment (minachuong) [inline code-challenges.js:74]: you can also use implicit returns. (general variable naming conventions leave out articles like `the`
```suggestion
   return arrayValue.map(value => value * 5)
```

File: code-challenges.js [hunk 6]
```diff
@@ -9,67 +9,171 @@
...
+// we want a function that will determine if each letter in the string is a vowel
+// the function will then remove all the vowels from the string
+
+// learned from google: .replace() -> regular expression that acts on strings to perform text search and text replace.
+
 var stringWithVowels1 = "HeyThereLearnStudent"
 var stringWithVowels2 = "ILoveJavaScript"
 
+const determines = (string) => {
+    var newString = string.replace(/[aeiou]/ig, "") // i makes the matching case insensitive, g finds all matches, and " " is the container for what .replace will replace the character match with, in this case nothing
```
Comment (minachuong) [inline code-challenges.js:95]: very cool finding a regex solution and explaining what all parts of the regex pattern are! However, using regex for a problem like this is a bit overkill. Consider how you might achieve the same outcome without regex.

File: code-challenges.js [hunk 7]
```diff
@@ -9,67 +9,171 @@
...
 
 
 
 // --------------------5) Copy/paste your code from #4. Refactor your code to include non-string edge cases. Inform your user that the variable passed into the vowel removal function is not a string.
 // Use the test variables provided below. Expected output: "true is not a string" "42 is not a string" "mCdngMstr"
 
+// we want to pass a value into our vowel remover and determine the type a value is
+// output if the typeOf the value is not a string if false
+// typeOf ---- return (variable) is not a string
+
```
Comment (minachuong) [inline code-challenges.js:112]: keep up the pseudocoding. For problems that require a lot of iterating, try write out a set of instructions for just the first value in the string to determine if the character is a vowel, then see if that logic works for the next character and so forth

File: code-challenges.js [hunk 8]
```diff
@@ -9,67 +9,171 @@
...
 var utensils = ["fork", "knife", "tongs", "chopsticks", "skewer"]
 var vessels = ["cup", "mug", "stemware", "glass", "tumbler", "tea cup"]
 
-
-
-
+const mostCharacters = (theArray) => {
+    let  result = " "; // take placeholder for new string (b)
+    for (let i=0; i < theArray.length; i++){ // iterate over length of array
```
Comment (minachuong) [inline code-challenges.js:145]: consider a higher order function that does what this for-loop is doing but in a declarative way (without having to manage the index)

**Review by minachuong — APPROVED** [review]
keep up the work on descriptive variable and function naming. Also the pseudocode was helpful! Awesome research on discovering regex however, consider how you would achieve the same outcome without it, but I also really appreciated the pseudocode explain regex because that's exactly what you'll likely need to do in the professional setting
